Framework/Algorithms/src/PolarizationCorrections/FlipperEfficiency.cpp

```diff
@@ -0,0 +1,127 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include <filesystem>
+
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidAlgorithms/PolarizationCorrections/FlipperEfficiency.h"
+#include "MantidAlgorithms/PolarizationCorrections/PolarizationCorrectionsHelpers.h"
+#include "MantidAlgorithms/PolarizationCorrections/SpinStateValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/Unit.h"
+
+namespace {
+/// Property Names
+namespace PropNames {
+auto constexpr INPUT_WS{"InputWorkspace"};
+auto constexpr OUTPUT_WS{"OutputWorkspace"};
+auto constexpr OUTPUT_FILE{"OutputFilePath"};
+auto constexpr SPIN_STATES{"SpinStates"};
+} // namespace PropNames
+
+auto constexpr FILE_EXTENSION{".nxs"};
+auto constexpr INITIAL_SPIN{"11,10,01,00"};
+} // namespace
+
+namespace Mantid::Algorithms {
+
+using namespace API;
+using namespace Kernel;
+using PolarizationCorrectionsHelpers::workspaceForSpinState;
+
+// Register the algorithm in the AlgorithmFactory
+DECLARE_ALGORITHM(FlipperEfficiency)
+
+std::string const FlipperEfficiency::summary() const { return "Calculate the efficiency of the polarization flipper."; }
+
+void FlipperEfficiency::init() {
+  declareProperty(std::make_unique<WorkspaceProperty<WorkspaceGroup>>(PropNames::INPUT_WS, "", Direction::Input),
+                  "Group workspace containing flipper transmissions for all 4 polarisation states.");
```

Comment:
Sorry, very minor and I should have spotted it before, but could we spell polarisation here with a z for consistency with all of our other user-facing uses of the word? Otherwise the rest of the changes are looking great and this will be good to go in after tweaking the spelling here.